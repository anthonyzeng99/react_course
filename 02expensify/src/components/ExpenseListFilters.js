import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/filters';
import { DateRangePicker } from 'react-dates';

class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  }
  onDatesChange = ({startDate, endDate}) => {
      this.props.dispatch(setStartDate(startDate));
      this.props.dispatch(setEndDate(endDate));
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({calendarFocused}));
  }
  render() {
    return (
      <div>
        <input
          type="text"
          defaultValue={this.props.filters.text}
          onChange={(e) => {
            dispatch(setTextFilter(e.target.value));
            console.log(e.target.value);
          }}
        />

        <select
          defaultValue={this.props.filters.sortBy}
          onChange={(e) => {
            if (e.target.value === 'date') {
              dispatch(sortByDate());
            } else if (e.target.value === 'amount') {
              dispatch(sortByAmount());
            }
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
};

export default connect(mapStateToProps)(ExpenseListFilters);
