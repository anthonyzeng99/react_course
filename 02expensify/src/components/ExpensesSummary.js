import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectVisibleExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedTotal = numeral(expensesTotal / 100).format('$0,0.00')

  return (
    <div>
      <h3>Viewing {expenseCount} {expenseWord} totaling {formattedTotal}</h3>
    </div>
  );
}

const mapStateToProps = (state) => {
  const visibleExpenses = selectVisibleExpenses(state.expenses, state.filters);
  const expensesTotal = selectExpensesTotal(state.expenses);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
