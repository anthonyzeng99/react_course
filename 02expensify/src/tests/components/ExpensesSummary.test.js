import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import { filters } from '../fixtures/filters';

test('should render ExpensesSummary correctly with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={100} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={25} expenseTotal={187523423} />);
  expect(wrapper).toMatchSnapshot();
});
