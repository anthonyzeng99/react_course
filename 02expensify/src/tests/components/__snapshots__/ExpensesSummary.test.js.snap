// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h3>
    Viewing 
    25
     
    expenses
     totaling 
    $0.00
  </h3>
</div>
`;

exports[`should render ExpensesSummary correctly with one expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totaling 
    $1.00
  </h3>
</div>
`;
